src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -488,6 +488,18 @@ const vector<Real>& SpatialPooler::getBoostedOverlaps() const
   return boostedOverlaps_;
 }
 
+/** helper method that checks SP output is stable for given configuration */
+bool checkUnstableParams_(SpatialPooler &sp) {
```

Comment:
this is the "stability check" that I suggest to run on each SP initialization. So far it has already identified many misconfigurations in our unit-tests.